File: app/src/main/java/getyourcasts/jd/com/getyourcasts/viewmodel/EpisodeState.java
package getyourcasts.jd.com.getyourcasts.viewmodel;

/**
 * Created by chuondao on 8/10/17.
 */

public class EpisodeState{
    private String uniqueId;
    private int state;
    private long transId;

    public EpisodeState(String uniqueId, int state, long transId) {
        this.uniqueId = uniqueId;
        this.state = state;
        this.transId = transId;
    }

    public static final int FETCHED = 0;
    public static final int DOWNLOADING = 2;
    public static final int DOWNLOADED = 1;
    public static final int DELETED = 3;

    public String getUniqueId() {
        return uniqueId;
    }

    public int getState() {
        return state;
    }

    public long getTransId() {
        return transId;
    }

    public void setUniqueId(String uniqueId) {
        this.uniqueId = uniqueId;
    }

    public void setState(int state) {
        this.state = state;
    }

    public void setTransId(long transId) {
        this.transId = transId;
    }
}

File: app/src/main/java/getyourcasts/jd/com/getyourcasts/viewmodel/EpisodeState.kt
package getyourcasts.jd.com.getyourcasts.viewmodel

/**
 *  class to help updating Episode State
 *  downloadLprogress should only be available when episode is
 *  DOWNLOADING otherwise -1 will be set
 */

class EpisodeState(var uniqueId: String?, var state: Int, val downloadProgress: Int) {
    companion object {
        const val EPISODE_FETCHED = 0
        const val EPISODE_DOWNLOADING = 2
        const val EPISODE_DOWNLOADED = 1
        const val EPISODE_DELETED = 3
    }
}
